src/Snapshotting/SnapshottingBehaviour.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace EventSauce\EventSourcing\Snapshotting;
+
+use EventSauce\EventSourcing\AggregateRoot;
+use EventSauce\EventSourcing\AggregateRootId;
+use Generator;
+
+trait SnapshottingBehaviour
+{
+    abstract public function aggregateRootVersion(): int;
+    abstract public function aggregateRootId(): AggregateRootId;
+    abstract protected function apply(object $event): void;
+
+    public function createSnapshot(): Snapshot
+    {
+        return new Snapshot($this->aggregateRootId(), $this->aggregateRootVersion(), $this->createSnapshotState());
+    }
+
+    abstract protected function createSnapshotState();
+
+    /**
+     * @param Snapshot  $snapshot
+     * @param Generator $events
+     *
+     * @return static
+     */
```

Comment:
Remove this DocBlock.

Reply:
This docblock is needed for PHPStan, the static part is inherited and then the type resolving works. I've kept the other params to not have half a docblock (which then gives IDE wrinkly's).

Reply:
Ah I see.